dandiapi/zarr/views/__init__.py

```diff
@@ -138,10 +139,14 @@ def create(self, request):
         if dandiset.embargo_status != Dandiset.EmbargoStatus.OPEN:
             raise ValidationError('Cannot add zarr to embargoed dandiset')
         zarr_archive: ZarrArchive = ZarrArchive(name=name, dandiset=dandiset)
-        try:
-            zarr_archive.save()
-        except IntegrityError as e:
-            raise ValidationError('Zarr already exists') from e
+        with transaction.atomic():
+            try:
```

Comment:
Just to clear up why this is done this way

```suggestion
        with transaction.atomic():
            # Use nested transaction block to prevent zarr creation race condition
            try:
```